var/spack/repos/builtin/packages/qscintilla/package.py

```diff
@@ -72,92 +70,68 @@ def setup_run_environment(self, env):
-                    "--sip-incdir="
-                    + join_path(
-                        self.spec["py-sip"].prefix.include,
-                        "python" + str(self.spec["python"].version.up_to(2)),
-                    ),
-                    "--stubsdir=" + pydir,
-                )
-
-                # Fix build errors
-                # "QAbstractScrollArea: No such file or directory"
-                # "qprinter.h: No such file or directory"
-                # ".../Qsci.so: undefined symbol: _ZTI10Qsci...."
-                qscipro = FileFilter("Qsci/Qsci.pro")
-                if "^qt@4" in self.spec:
-                    qtx = "qt4"
-                elif "^qt@5" in self.spec:
-                    qtx = "qt5"
-
-                link_qscilibs = "LIBS += -L" + self.prefix.lib + " -lqscintilla2_" + qtx
-                qscipro.filter(
-                    "TEMPLATE = lib",
-                    "TEMPLATE = lib\nQT += widgets" + "\nQT += printsupport\n" + link_qscilibs,
-                )
-
-                make()
-
-                # Fix installation prefixes
-                makefile = FileFilter("Makefile")
-                makefile.filter(r"\$\(INSTALL_ROOT\)", "")
-                makefile = FileFilter("Qsci/Makefile")
-                makefile.filter(r"\$\(INSTALL_ROOT\)", "")
-
-                if "@2.11:" in self.spec:
-                    make("install", parallel=False)
-                else:
-                    make("install")
-
-    @run_after("install")
-    def extend_path_setup(self):
-        if self.spec["py-sip"].satisfies("@:4"):
-            # See github issue #14121 and PR #15297
-            module = self.spec["py-sip"].variants["module"].value
-            if module != "sip":
-                module = module.split(".")[0]
-                with working_dir(python_platlib):
-                    with open(os.path.join(module, "__init__.py"), "w") as f:
-                        f.write("from pkgutil import extend_path\n")
-                        f.write("__path__ = extend_path(__path__, __name__)\n")
+        with working_dir(os.path.join(self.stage.source_path, "designer")):
+            # TODO: qmake fails with qt6
+            qmake('designer.pro', 'INCLUDEPATH+=../src')
+            make()
+            makefile = FileFilter("Makefile")
+            makefile.filter(r"\$\(INSTALL_ROOT\)" + self.spec["qt"].prefix, "$(INSTALL_ROOT)")
+            make("install")
+
+    @run_after("install", when='+python')
+    def make_qsci_python(self):
+        if "^py-pyqt5" in self.spec:
+            py_pyqtx = "py-pyqt5"
+            pyqtx = "PyQt5"
+            ftoml = 'pyproject-qt5.toml'
+        elif "^py-pyqt6" in self.spec:
+            py_pyqtx = "py-pyqt6"
+            pyqtx = "PyQt6"
+            ftoml = 'pyproject-qt6.toml'
+
+        with working_dir(join_path(self.stage.source_path, "Python")):
+            cp = which('cp')
+            cp(ftoml, 'pyproject.toml')
```

Comment:
This will work on Windows too
```suggestion
            copy(ftoml, 'pyproject.toml')
```